### PR TITLE
Fix g++-13 build errors

### PR DESCRIPTION
**Pull request by cyyever**
### Description

By adding -Wno-stringop-overflow

### CODE REVIEW

**Comment by cyyever** [comment]
@justinchuby And this one

File: .github/workflows/main.yml
Comment (andife) [inline .github/workflows/main.yml:1]: If we take the standard version that comes with ubuntu 24.04, we don't have to install g++ separately

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -70,13 +70,10 @@ jobs:
...
           cmake --version
 
-      # 2024.10.15: This step can be adjusted or even deleted if we decide to use a higher G++ or protobuf version.
-      # With the current protobuf version, g++13 does not compile due to missing headers
       - name: Show versions g++
         if: startsWith(matrix.os,'ubuntu')
         run: |
-          sudo apt install g++-11
-          sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 12
+          sudo apt install g++
```
Comment (justinchuby) [inline .github/workflows/main.yml:76]: Do we still need to install, or can we use the stock version ?

Comment (cyyever) [reply .github/workflows/main.yml:76]: I think we can use the stock version

File: CMakeLists.txt [hunk 2]
```diff
@@ -650,6 +650,10 @@ if(MSVC)
                                    "${onnx_static_library_flags}")
 else()
   target_compile_options(onnx PRIVATE -Wall -Wextra)
+  if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 13)
+    target_compile_options(onnx PRIVATE "-Wno-stringop-overflow")
+    target_compile_options(onnx_proto PRIVATE "-Wno-stringop-overflow")
```
Comment (justinchuby) [inline CMakeLists.txt:655]: Do you know if this can be resolved eventually?

**Review by justinchuby — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (cyyever) [reply CMakeLists.txt:655]: They are protobuf bugs. To be honest, protobuf is far from warning free...

**Pull request merged**